Composer/packages/electron-server/src/utility/machineId.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import crypto from 'crypto';
+
+import { v4 as uuid } from 'uuid';
+
+import { getMacAddress } from './getMacAddress';
+
+export async function getMachineId(): Promise<string> {
+  return (await getMacMachineId()) || uuid.generateUuid(); // fallback, generate a UUID
```

Comment:
I'll review more thoroughly soon but here's a quick comment:
I think the very first time this id should be generated and persisted and later on just use the persisted id.
(Check persisted file for ```telemetryId```, if it doesn't exist, generate one: it's gonna be either the mac address or a uuid. Later on, use the persisted ```telemetryId```)